### ISSUE TITLE
Separate example project from test project

### ISSUE THREAD

**Issue by marco-m** [RickMoynihan/lein-tools-deps#10]
See this conversation https://github.com/RickMoynihan/lein-tools-deps/pull/8 about `.lein-classpath` and the `example` project. Example should not have `.lein-classpath`. See leiningen doc at https://github.com/technomancy/leiningen/blob/stable/doc/PLUGINS.md#local-development

**Comment by RickMoynihan** [comment]
I'd rather not do this anytime soon and have to maintain two almost identical things, especially when almost nothing works yet :-)

**Comment by marco-m** [comment]
The way I was thinking about it was:
1. remove `.lein-classpath` from example project
2. in the test, use example project as template and add `.lein-classpath` from the test code itself

I am used to fully automated, start-from-scratch tests, since it is the only way to ensure reproducibility. In other words, treat tests with the same "immutable" idea of FP itself. Tests can go both on travis (which always start from scratch) and on Docker, to allow running locally and starting from scratch.

But the project is yours, do as you please :-)

**Comment by RickMoynihan** [comment]
Yes, I'm not against automated repeatable testing at all, and would definitely be up for doing something that.  This particular ticket didn't imply that was your intention though, instead I saw it as removing a piece of our only current dev/testing mechanism!

However the example project is (for me at least) in a broken state that I've not yet been able to fix, i.e. it doesn't seem to resolve its dependencies upon a `lein deps`.

Things worked for me when I first published the project, but I think `tools.deps` at that point were on a `SNAPSHOT` commit and hadn't actually published a jar for it.  So I was forced to `lein install` one locally.  Since then I think they _may_ have published a new `SNAPSHOT` which has led to some breakage for me.

Other people seem to have things working though, but no longer me.  And I've not figured out why yet.  It may be because of other changes, and I suspect isolating/installing from scratch in travis is a good way to help tease these issues out.  Especially as the `clj` command line tools are also moving targets independently of the tools.deps library.

Anyway since then there have been a few updates to tools.deps which also broke the API etc, but they did actually release a real artifact, so I updated things to lock against the newest and fixed the most obvious API breakages in this commit: b0099462466c4c2d0c2c7855fd3265fa4ab10579.  

However it still hasn't resolved things for me.

Anyway I'm happy to reopen this ticket to create a meaningful test.

As a starter for 10 I'd suggest something like the following process for travis/test:

- `apt-get install clojure lein` etc (travis)
- `lein test`
  - test then copies example to a `tmp` directory and adds a `.lein-classpath` to it (setup)
  - test then shells out running `lein deps` inside example project (strictly not necessary but good to run it separetely to aid understanding failures).  This should pull criterium or another well known dep.
  - test then does `lein run` in example which executes `-main`.
  - `-main` calls `(require 'criterium.core) (println "done")`
  - test captures stdout/stderr and expects to see the string `"done"`

Obviously we could improve but I think that would be a pretty good smoke test to start with, what do you think?

**Comment by RickMoynihan** [comment]
Added some crude unit tests, that now use standalone `deps.edn` files that aren't the example one.  Example is currently used for manual integration tests, though we should get travis to do that too.
